app/modules/users/models.py

```diff
@@ -526,6 +527,43 @@ def has_permission_to_read(self, obj):
 
         return has_permission
 
+    # Permissions control entry point for real users, for all objects and all operations
+    def can_perform_action(self, obj, action):
+        has_permission = False
+        if action == operation.ObjectAccessOperation.READ:
+            if obj is None:
+                # @todo, this is for the "clone if not known" functionality that is under debate
+                has_permission = True
+            else:
+                has_permission = self._has_permission_to_read(obj)
+        elif action == operation.ObjectAccessOperation.WRITE:
+            if obj is None:
+                # Allowed to write (create) an object that doesn't exist
+                has_permission = True
+            else:
+                has_permission = self.owns_object(obj)
+                # @todo this is where project and collaborations would link in
+        elif action == operation.ObjectAccessOperation.DELETE:
+            if obj is None:
+                has_permission = False
+            else:
+                has_permission = self.owns_object(obj)
+                # @todo this is where project and collaborations would link in
+
+        return has_permission
+
+    # Permissions control entry point for anonymous users, for all objects and all operations
+    @classmethod
+    def anonymous_can_perform_action(cls, obj, action):
+        # Anonymous users can only read public objects
+        has_permission = (
+            action == operation.ObjectAccessOperation.READ
+            and callable(hasattr(obj, 'is_public'))
```

Comment:
Let's elevate this `is_public` class attribute to a default value in a `HoustonModel` so that this existence check with `hasattr` will always be True.  We can overwrite the implementation for `is_public` for the `Encounter` model to be the correct logic, whereas the HoustonModel definition will always default to `False`

Reply:
I actually put the implementation in FeatherModel as not everything inherits from HoustonModel